doc/api/v8.md

```diff
@@ -1051,6 +1051,51 @@ added:
 
 Returns true if the Node.js instance is run to build a snapshot.
 
+### Class: `v8.MemoryProfileWatchdog`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+#### `new MemoryProfileWatchdog([options])`
+
+* `interval` {number} How often the watchdog polls.
+* `maxRss` {number} Node.js will take snapshot when rss is
+  greater or equla than `maxRss`.
+* `maxUsedHeapSize` {number} Node.js will take snapshot when used heap size is
+  greater or equla than `maxUsedHeapSize`.
+* `filename` {string} Node.js will write heap snapshot to `filename`.
+
+Creates a new `MemoryProfileWatchdog` object, `maxRss` or `maxUsedHeapSize` must
+be setted. `MemoryProfileWatchdog` will create a thread and poll the memory
+info periodically, when the rss or used heap size is greater or equla than
+the value you set, Node.js will write a heap snapshot to `filename` then stop
+the watchdog.
```

Comment:
Too much policy, too little mechanism, IMO. What I mean is: it does one thing and one thing only. It's inflexible.

Also: equla -> equal

Reply:
Thanks for your suggestions, my initial idea was to check the memory, cpu load and so on in a thread, and then change it to do one thing per thread. What you mean is the mechanism is too simple ? Can you give me some specific suggestions ? Thank you very much !

Reply:
"Mechanism, not policy" means you give the user the building blocks to accomplish a task but you don't put those blocks together in a preset configuration. It's the distinction between a toolkit and a framework.

The reason is that you can never anticipate all possible use cases users are going to come up with. It also means that otherwise perfectly reasonable functionality (like this watchdog) may not be suitable for inclusion in core because it's not malleable enough.

Reply:
My idea is to provide some specific functions, not a watchdog mechanism, but these functions are implemented through the watchdog 😄.